flink-clients/src/main/java/org/apache/flink/client/program/Client.java

```diff
@@ -65,31 +73,25 @@
 /**
  * Encapsulates the functionality necessary to submit a program to a remote cluster.
  */
-public class Client {
+public abstract class Client {
```

Comment:
I wonder if we can rid of some of the overloaded methods here.

For example
```
public JobGraph getJobGraph(PackagedProgram prog, FlinkPlan optPlan) throws ProgramInvocationException
public JobGraph getJobGraph(PackagedProgram prog, FlinkPlan optPlan, String savepointPath) throws ProgramInvocationException
```

seem both unused (on first sight, not sure).

![image](https://cloud.githubusercontent.com/assets/89049/15355314/301e8e9a-1cf2-11e6-84ef-592609320f9f.png)


Ideally we have one runBlocking, one runDetached and one plan optimization. I'm not sure if its really necessary to have so many variants ;)